rest/src/main/java/org/jboss/pnc/rest/provider/BuildRecordProvider.java

```diff
@@ -660,63 +696,76 @@ public BuildRecordRest getLatestBuildRecord(Integer configId, boolean executedOn
      * complete. This abstraction allows clients to query for a list of all builds whether running or completed.
      */
     private CollectionInfo<BuildRecordRest> getBuilds(Integer pageIndex, Integer pageSize, String sort, String orFindByBuildConfigurationName, String andFindByBuildConfigurationName, String... rsqlQueries) {
+
+        logger.debug(
+                "Get running and completed build records by pageIndex: {}, pageSize: {}, sort: {}, orFindByBuildConfigurationName: {}, andFindByBuildConfigurationName: {}, rsqlQueries: {}",
+                pageIndex, pageSize, sort, orFindByBuildConfigurationName, andFindByBuildConfigurationName, rsqlQueries);
+
+        // Combine predicates of both search query and the additional filters (e.g.userId, buildConfigurationId)
         List<Predicate<BuildRecord>> dbAndPredicatesList = Arrays.stream(rsqlQueries)
                 .filter(rsqlString -> rsqlString != null && !rsqlString.isEmpty())
                 .map(p -> rsqlPredicateProducer.getPredicate(BuildRecord.class, p))
                 .collect(Collectors.toList());
         List<Predicate<BuildRecord>> dbOrPredicateList = new ArrayList<>();
 
+        // Combine rsqlQueries AND-ing all the single queries
         String combinedQueries = combineRsqlQueriesMatchAll(rsqlQueries);
-        if (!StringUtils.isEmpty(orFindByBuildConfigurationName)) {
-            //add steam condition
+
+        // An AND condition (andFindByBuildConfigurationName) gets higher precedence over an OR condition
+        // (orFindByBuildConfigurationName). If specified both, only AND is applied.
+
+        if (!StringUtils.isEmpty(andFindByBuildConfigurationName)) {
+            // add steam condition
+            StringBuilder sb = new StringBuilder(";(buildConfigurationName==").append(andFindByBuildConfigurationName).append(")");
+
             if (StringUtils.isEmpty(combinedQueries)) {
-                combinedQueries = "(buildConfigurationName=like=" + orFindByBuildConfigurationName + ")";
+                combinedQueries = sb.deleteCharAt(0).toString();
             } else {
-                combinedQueries = "(" + combinedQueries + "),(buildConfigurationName=like=" + orFindByBuildConfigurationName + ")";
+                combinedQueries = sb.insert(0, ")").insert(0, combinedQueries).insert(0, "(").toString();
```

Comment:
You don't need to use StringBuilder explicitly, the compiler will use it for you and the code will be nicer (e.g. not reversed order).

https://dzone.com/articles/string-concatenation-performacne-improvement-in-ja

Reply:
Thanks for the hint, I guess I haven't been coding enough lately :-). And honestly my version looked so bad! :-)